packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -191,22 +187,55 @@ export interface ContainerRuntimeMessage {
     contents: any;
     type: ContainerMessageType;
 }
+export interface ISummaryConfigurationBaseSettings {
+    /* Delay before first attempt to spawn summarizing container. */
+    initialSummarizerDelayMs: number;
+
+    /**
+     * Flag that will enable changing elected summarizer client after maxOpsSinceLastSummary.
+     * This defaults to false (disabled) and must be explicitly set to true to enable.
+     */
+    summarizerClientElection: boolean;
+
+    maxAckWaitTime: number;
+    maxOpsSinceLastSummary: number;
+}
+
+export interface ISummaryConfigurationHeuristicSettings extends ISummaryConfigurationBaseSettings {
+    idleTime: number;
+    maxTime: number;
+    maxOps: number;
+    minOpsForAttemptOnClose: number;
+}
+
+export type ISummaryConfiguration =
+{
+    state: "disabled";
+}
+| ({ state: "disableHeuristics"; } & ISummaryConfigurationBaseSettings)
+| ({ state: "enabled";} & ISummaryConfigurationHeuristicSettings);
 
 // Consider idle 5s of no activity. And snapshot if a minute has gone by with no snapshot.
 const IdleDetectionTime = 5000;
 
 const DefaultSummaryConfiguration: ISummaryConfiguration = {
+    state: "enabled",
     idleTime: IdleDetectionTime,
 
     maxTime: IdleDetectionTime * 12,
 
     // Snapshot if 1000 ops received since last snapshot.
     maxOps: 1000,
 
+    minOpsForAttemptOnClose: 50,
     // Wait 2 minutes for summary ack
     // this is less than maxSummarizeAckWaitTime
     // the min of the two will be chosen
     maxAckWaitTime: 120000,
+
+    maxOpsSinceLastSummary: 7000,
+    initialSummarizerDelayMs: 5000,
+    summarizerClientElection: false,
 };
```

Comment:
Love to see all of them in one place :) I'm sure there are more things hidden in the code like that :)